src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -398,8 +406,39 @@ public SemanticVersion(string version)
 
             Major = v.Major;
             Minor = v.Minor;
-            Patch = v.Patch;
-            Label = v.Label;
+            Patch = v.Patch < 0 ? 0 : v.Patch;
+            PreReleaseLabel = v.PreReleaseLabel;
+            BuildLabel = v.BuildLabel;
+        }
+
+        /// <summary>
+        /// Construct a SemanticVersion.
+        /// </summary>
+        /// <param name="major">The major version</param>
+        /// <param name="minor">The minor version</param>
+        /// <param name="patch">The patch version</param>
+        /// <param name="preLabel">The preLabel for the version</param>
+        /// <param name="buildLabel">The buildLabel for the version</param>
+        /// <exception cref="FormatException">
+        /// If <paramref name="preLabel"/> don't match 'LabelUnitRegEx'.
+        /// If <paramref name="buildLabel"/> don't match 'LabelUnitRegEx'.
+        /// </exception>
+        public SemanticVersion(int major, int minor, int patch, string preLabel, string buildLabel)
```

Comment:
Shall we rename `preLabel` to `prereleaseLabel` too? It's the parameter of a public method.

Reply:
Renamed.